scripts/rotate-aws-creds/sharded.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+set -euo pipefail
+
+source "$(dirname $0)/common.sh"
+
+SHARD_SSH="ssh root@hive-production-master"
+OC_CMD="oc --insecure-skip-tls-verify=true"
+
+fixup_kubeconfig() {
+    sed -i 's#https://api.#https://rh-api.#' "$1"
+}
+
+cleanup() {
+    rm -f "${SHARD_KUBECONFIG}"
+    rm -f "${CLUSTER_KUBECONFIG}"
+    rm -r "${CREDREQ_DIR}"
+}
+
+update_aws_secret() {
+    local KUBECONFIG=$1
+    local SECRET_NAMESPACE=$2
+    local SECRET_NAME=$3
+    # assumes encoded vars are set already by iam_user_create_accesskey
+
+    KUBECONFIG=${KUBECONFIG} ${OC_CMD} get secret -n "${SECRET_NAMESPACE}" "${SECRET_NAME}" -o json | jq -r ".data.aws_access_key_id = \"${ENCODED_ACCESS_KEY_ID}\" | .data.aws_secret_access_key = \"${ENCODED_SECRET_ACCESS_KEY}\"" | KUBECONFIG=${KUBECONFIG} ${OC_CMD} replace -f -
+}
+
+CLUSTER_NAME=${1:-}
+SHARD_NAME=${2:-}
+if [[ ${CLUSTER_NAME} == "" || ${SHARD_NAME} == "" ]]; then
+    echo "Usage: $0 CLUSTER_NAME SHARD_NAME"
+    exit 1
+fi
+
+CLUSTER_ID=$(ocm list clusters | grep "${CLUSTER_NAME}" | awk '{print $1}')
+
+SHARD_KUBECONFIG=$(mktemp /tmp/kubeconfig.XXXXXXX)
+CLUSTER_KUBECONFIG=$(mktemp /tmp/kubeconfig.XXXXXXX)
+CREDREQ_DIR=$(mktemp -d)
+trap cleanup exit 
+
+SHARD_NAMESPACE=$(${SHARD_SSH} oc get accountclaims --all-namespaces | grep "${SHARD_NAME}" | awk '{print $1}')
+SHARD_KUBECONFIG_SECRET=$(${SHARD_SSH} oc get secret -n "${SHARD_NAMESPACE}" | grep kubeconfig | awk '{print $1}')
+${SHARD_SSH} "oc get secret -n ${SHARD_NAMESPACE} ${SHARD_KUBECONFIG_SECRET} -o json" | jq -r '.data["raw-kubeconfig"]' | base64 -d >"${SHARD_KUBECONFIG}"
+fixup_kubeconfig "${SHARD_KUBECONFIG}"
+
+CLUSTER_NAMESPACE=$(KUBECONFIG=${SHARD_KUBECONFIG} ${OC_CMD} get accountclaims --all-namespaces | grep "${CLUSTER_ID}" | awk '{print $1}')
+CLUSTER_KUBECONFIG_SECRET=$(KUBECONFIG=${SHARD_KUBECONFIG} ${OC_CMD} get secret -n "${CLUSTER_NAMESPACE}" | grep kubeconfig | awk '{print $1}')
+KUBECONFIG="${SHARD_KUBECONFIG}" ${OC_CMD} get secret -n "${CLUSTER_NAMESPACE}" "${CLUSTER_KUBECONFIG_SECRET}" -o json | jq -r '.data["raw-kubeconfig"]' | base64 -d >"${CLUSTER_KUBECONFIG}"
+fixup_kubeconfig "${CLUSTER_KUBECONFIG}"
+
+CCS_ANNOTATION=$(KUBECONFIG="${SHARD_KUBECONFIG}" ${OC_CMD} get clusterdeployment -n "${CLUSTER_NAMESPACE}" "${CLUSTER_NAME}" -o json | jq -r '.metadata.labels["api.openshift.com/ccs"]')
+if [[ "${CCS_ANNOTATION}" == "true" ]]; then
+    CCS_CLUSTER="1"
+    CREDENTIALS_SECRET_NAME="byoc"
```

Comment:
There is also an `aws` secret for CCS.  Do we need to rotate it?

Reply:
When I first implemented this script the secret had a different name. Since then things might have changed. Maybe @fahlmant might know more.